addons/web_editor/static/src/components/media_dialog/document_selector.js

```diff
@@ -0,0 +1,74 @@
+/** @odoo-module */
+
+import { Attachment, FileSelector, IMAGE_MIMETYPES } from './file_selector';
+
+export class DocumentAttachment extends Attachment {}
+DocumentAttachment.template = 'web_editor.DocumentAttachment';
+
+export class DocumentSelector extends FileSelector {
+    setup() {
+        super.setup();
+
+        this.uploadText = this.env._t("Upload a document");
+        this.urlPlaceholder = "https://www.odoo.com/mydocument";
+        this.addText = this.env._t("Add document");
+        this.searchPlaceholder = this.env._t("Search a document");
+        this.allLoadedText = this.env._t("All documents have been loaded");
+    }
+
+    get attachmentsDomain() {
+        const domain = super.attachmentsDomain;
+        domain.push(['mimetype', 'not in', IMAGE_MIMETYPES]);
+        // The assets should not be part of the documents.
+        // All assets begin with '/web/assets/', see _get_asset_template_url().
+        domain.unshift('&', '|', ['url', '=', null], '!', ['url', '=like', '/web/assets/%']);
+        return domain;
+    }
+
+    async onClickDocument(document) {
+        this.selectAttachment(document);
+        await this.props.save();
+    }
+
+    async fetchAttachments(...args) {
+        const attachments = await super.fetchAttachments(...args);
+
+        if (this.selectInitialMedia()) {
+            for (const attachment of attachments) {
+                if (`/web/content/${attachment.id}` === this.props.media.getAttribute('href').replace(/[?].*/, '')) {
+                    this.selectAttachment(attachment);
+                }
+            }
+        }
+        return attachments;
+    }
+
+    /**
+     * Utility method used by the MediaDialog component.
+     */
+    static async createElements(selectedMedia, { orm }) {
+        return Promise.all(selectedMedia.map(async attachment => {
+            const linkEl = document.createElement('a');
+            let href = `/web/content/${attachment.id}?unique=${attachment.checksum}&dowload=true`;
+            if (!attachment.public) {
+                const [accessToken] = await orm.call(
+                    'ir.attachment',
+                    'generate_access_token',
+                    [attachment.id],
+                );
+                href += `&access_token=${accessToken}`;
```

Comment:
For the record, this is relying on the unspecified token format, which could turn out not to be safe at some point. Dynamic URL path components and query parameters should be url-encoded as a general rule (i.e. escaped).

Reply:
As discussed, we'll track down the occurrences of such mistakes in stable (that one here already existed before this PR) and forward-port correctly up to master. I am adding an entry in our bug-fix pad.